acceptance_test.go

```diff
@@ -0,0 +1,279 @@
+		}
+		credFilePath = credFile.Name()
+	} else {
+		t.Error("credentials not set in env CONDUIT_GOOGLE_CREDENTIAL_JSON")
+		t.FailNow()
+	}
+
+	tokenJSON := strings.TrimSpace(os.Getenv("CONDUIT_GOOGLE_TOKEN_JSON"))
+	if tokenJSON != "" {
+		tokenFile, err := os.CreateTemp("", "token*.json")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.Remove(tokenFilePath)
+
+		if _, err = tokenFile.WriteString(tokenJSON); err != nil {
+			t.Error("error writing token file", err)
+		}
+		tokenFilePath = tokenFile.Name()
+	} else {
+		t.Error("token not set in env CONDUIT_GOOGLE_TOKEN_JSON")
+		t.FailNow()
+	}
+
+	sheetURL = strings.TrimSpace(os.Getenv("CONDUIT_GOOGLE_SHEET_URL"))
+	if sheetURL == "" {
+		t.Error("sheetURL not set in env CONDUIT_GOOGLE_SHEET_URL")
+		t.Skip()
+	}
+
+	sheetName = strings.TrimSpace(os.Getenv("CONDUIT_GOOGLE_SHEET_NAME"))
+	if sheetName == "" {
+		t.Error("sheetName not set in env CONDUIT_GOOGLE_SHEET_NAME")
+		t.FailNow()
+	}
+
+	sourceConfig := map[string]string{
+		"credentialsFile": credFilePath,
+		"tokensFile":      tokenFilePath,
+		"sheetsURL":       sheetURL,
+		"pollingPeriod":   "1s", // Configurable polling period
+	}
+
+	destConfig := map[string]string{
+		"credentialsFile":  credFilePath,
+		"tokensFile":       tokenFilePath,
+		"sheetsURL":        sheetURL,
+		"sheetName":        sheetName,
+		"valueInputOption": "USER_ENTERED",
+		"bufferSize":       "10",
+	}
+
+	ctx := context.Background()
+	conf, err := config.Parse(sourceConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	spreadsheetID = conf.GoogleSpreadsheetID
+	sheetID = conf.GoogleSheetID
+
+	client := conf.OAuthConfig.Client(ctx, conf.OAuthToken)
+	sheetService, err := sheets.NewService(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	clearSheet := func(t *testing.T) {
+		_, err := sheetService.Spreadsheets.Values.Clear(conf.GoogleSpreadsheetID, "1:1000", &sheets.ClearValuesRequest{}).Do()
+		if err != nil {
+			t.Errorf("error cleaning the sheet: %v", err.Error())
+		}
+	}
+
+	// clear sheet before starting the tests
+	clearSheet(t)
+	sdk.AcceptanceTest(t, AcceptanceTestDriver{
+		rand: rand.New(rand.NewSource(time.Now().UnixNano())), //nolint: gosec // only used for testing
+		ConfigurableAcceptanceTestDriver: sdk.ConfigurableAcceptanceTestDriver{
+			Config: sdk.ConfigurableAcceptanceTestDriverConfig{
+				Connector: sdk.Connector{
+					NewSpecification: Specification,
+					NewSource:        source.NewSource,
+					NewDestination:   destination.NewDestination,
+				},
+				SourceConfig:      sourceConfig,
+				DestinationConfig: destConfig,
+				BeforeTest: func(t *testing.T) {
+				},
+				GoleakOptions: []goleak.Option{goleak.IgnoreCurrent(), goleak.IgnoreTopFunction("internal/poll.runtime_pollWait")},
+				Skip: []string{
+					// skipped because it uses pre-written generateRecord function, which doesn't generate data in g sheets format
+					// leading to the test always failing
```

Comment:
Can you please elaborate a bit more on this? I see that there is a generateRecords function below.

Reply:
We have overwritten the WriteToSource helper function, but `TestDestination_WriteAsync_Success` directly generates the records using the predefined generateRecord function, which is not a valid format.

Reply:
Gotcha... We can probably find a way to handle that, but it doesn't pay out since we have https://github.com/ConduitIO/conduit-connector-sdk/issues/16 which should soon be done.